src/genomic_features/ensembl/ensembldb.py

```diff
@@ -27,13 +41,66 @@ def annotation(species: str, version: str | int):
     EnsemblDB
         The annotation database.
     """
-    return EnsemblDB(
-        ibis.sqlite.connect(
-            retrieve_annotation(
-                ENSEMBL_URL_TEMPLATE.format(species=species, version=version)
+    try:
+        ensdb = EnsemblDB(
+            ibis.sqlite.connect(
+                retrieve_annotation(
+                    ENSEMBL_URL_TEMPLATE.format(species=species, version=version)
+                )
             )
         )
-    )
+    except HTTPError as err:
+        if err.response.status_code == 404:
+            raise ValueError(
+                f"No Ensembl database found for {species} v{version}. Check available versions with `genomic_features.ensembl.list_versions`."
+            ) from err
+        else:
+            raise HTTPError from err
+    return ensdb
+
+
+def list_versions(species: str) -> DataFrame:
+    """List available Ensembl versions for a species.
+
+    Parameters
+    ----------
+    species
+        The species name. E.g. Hsapiens for human, Mmusculus for mouse.
+
+    Returns
+    -------
+    DataFrame
+        A table of available versions in EnsDb for species of interest.
+    """
+    ahdb = ibis.sqlite.connect(retrieve_annotation(ANNOTATION_HUB_URL))
+
+    # Get latest AnnotationHub timestamp
+    timestamp = requests.get(TIMESTAMP_URL).text
+    ahdb = ibis.sqlite.connect(retrieve_annotation(ANNOTATION_HUB_URL))
+    latest_ts = Timestamp(timestamp).replace(tzinfo=None)
+    cached_ts = ahdb.table("timestamp").execute()["timestamp"][0]
+    if latest_ts != cached_ts:
+        cached_db = glob.glob(
+            os.path.join(pooch.os_cache("genomic-features"), "*annotationhub.sqlite3")
+        )[0]
+        os.remove(cached_db)
+        ahdb = ibis.sqlite.connect(retrieve_annotation(ANNOTATION_HUB_URL))
```

Comment:
```suggestion
    # Get latest AnnotationHub timestamp
    db_path = retrieve_annotation(ANNOTATION_HUB_URL)
    timestamp = requests.get(TIMESTAMP_URL).text
    ahdb = ibis.sqlite.connect(db_path)
    latest_ts = Timestamp(timestamp).replace(tzinfo=None)
    cached_ts = ahdb.table("timestamp").execute()["timestamp"][0]
    if latest_ts != cached_ts:
        db_path.unlink()
        ahdb = ibis.sqlite.connect(retrieve_annotation(ANNOTATION_HUB_URL))
```

* removes an extra call to `connect` 
* deletes using pathlib